compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -222,6 +222,9 @@
     private static final String DEPRECATION_ANNOTATION = "deprecated";
     private static final String ANONYMOUS_RECORD_NAME = "anonymous-record";
 
+    private boolean simpleVarInIterativeUse;
+    private boolean simpleVarInFunctionParam;
```

Comment:
Instead of a SymbolEnter level state variable better to use a flag. Better to move the logic to type checker to a place we analyze function parameters?